ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/CreateSymbolStylesFromWebStyles/CreateSymbolStylesFromWebStyles.cpp

```diff
@@ -0,0 +1,166 @@
+// [WriteFile Name=CreateSymbolStylesFromWebStyles, Category=DisplayInformation]
+// [Legal]
+// Copyright 2021 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "CreateSymbolStylesFromWebStyles.h"
+
+#include "FeatureLayer.h"
+#include "ServiceFeatureTable.h"
+#include "Map.h"
+#include "MapQuickView.h"
+
+#include "UniqueValueRenderer.h"
+#include "SymbolStyle.h"
+#include "SimpleMarkerSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+
+CreateSymbolStylesFromWebStyles::CreateSymbolStylesFromWebStyles(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISNavigation, this))
+{
+  QUrl webStyleLayerUrl = QUrl("http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0");
+  m_webStyleLayer = new FeatureLayer(new ServiceFeatureTable(webStyleLayerUrl, this), this);
+
+  m_uniqueValueRenderer = new UniqueValueRenderer(this);
+
+  // The UniqueValueRenderer will affect specific features based on the values of the specified FieldName(s)
+  m_uniqueValueRenderer->setFieldNames({"cat2"});
+
+  // The UniqueValueRenderer defines how features of a FeatureLayer are stylized
+  // Without an overriding UniqueValueRenderer, features from m_webStyleLayer will use the web layer's default gray circle style
+  m_webStyleLayer->setRenderer(m_uniqueValueRenderer);
+
+  m_map->operationalLayers()->append(m_webStyleLayer);
+
+  // Populate the legend from the UniqueValuesRenderer's UniqueValuesList once all UniqueValues have been added
+  connect(m_map->legendInfos(), &LegendInfoListModel::fetchLegendInfosCompleted, this, [this]()
+  {
+    m_legendInfoListModel = m_map->legendInfos();
+    emit legendInfoListModelChanged();
+  });
+
+  // Set the scale at which feature symbols and text will appear at their default size
+  m_map->setReferenceScale(100'000);
+
+  createSymbolStyles();
+}
+
+void CreateSymbolStylesFromWebStyles::createSymbolStyles()
+{
+  m_categoriesMap = createCategoriesMap();
+  m_symbolsFetchedCount = 0;
+
+  for (const QString &symbolKey : m_categoriesMap.keys())
+  {
+    SymbolStyle* symbolStyle = new SymbolStyle("Esri2DPointSymbolsStyle", {}, this);
```

Comment:
I'm not sure that this is the optimal pattern to showcase. SymbolStyle is a container that has a bunch of symbols in it that you can fetch on demand. Instead of creating a new SymbolStyle instance every time you want a new symbol, wouldn't it make more sense to create one instance, and call fetchSymbol on that same instance for each key you want to fetch?

Reply:
I thought that would be the case too, but it doesn't work if we use that pattern.

Reply:
> but it doesn't work if we use that pattern.

That doesn't seem right. What doesn't work?

Reply:
can you move your signal connection to be outside/before the for loop? right under the object instantiation.